File: Python/392. Is Subsequence.py
# noinspection PyMethodMayBeStatic
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        length_s = len(s)
        length_t = len(t)

        if length_s > length_t:
            return False

        if length_s == 0:
            return True

        subsequence = 0

        for i in range(length_t):
            if s[subsequence] == t[i]:
                subsequence += 1

                if subsequence == length_s:
                    break

        return length_s == subsequence

File: TypeScript/392. Is Subsequence.ts
// noinspection JSUnusedGlobalSymbols
const isSubsequence = (s: string, t: string): boolean => {
  const lengthS = s.length
  const lengthT = t.length

  if (lengthS > lengthT) {
    return false
  }

  if (lengthS === 0) {
    return true
  }

  let subsequence = 0

  for (let i = 0; i < lengthT; i++) {
    if (s[subsequence] === t[i]) {
      subsequence++
    }
  }

  return lengthS === subsequence
}
